modules/github_integration/lib/open_project/github_integration/notification_handler/issue_comment.rb

```diff
@@ -44,10 +46,14 @@ def process(params)
 
         github_system_user = User.find_by(id: payload.open_project_user_id)
         work_packages = find_mentioned_work_packages(payload.comment.body, github_system_user)
-        new_work_packages = without_already_referenced(work_packages, pull_request)
 
-        upsert_partial_pull_request(new_work_packages)
-        comment_on_referenced_work_packages(new_work_packages, github_system_user, journal_entry) if new_work_packages.any?
+        new_work_packages = without_already_referenced(work_packages, pull_request&.work_packages.to_a || [])
+        return unless new_work_packages.any?
```

Comment:
(minor)
As I have difficulties reading `unless` statements, I tend to prefer `if` and would rewrite it like this:

```
return if new_work_packages.none?
```